recipes/fmt/all/conanfile.py

```diff
@@ -42,13 +41,19 @@ def _has_with_os_api_option(self):
     def generate(self):
         if not self.options.header_only:
             tc = CMakeToolchain(self)
-            tc.generate()  
+            tc.cache_variables["FMT_DOC"] = False
+            tc.cache_variables["FMT_TEST"] = False
+            tc.cache_variables["FMT_INSTALL"] = True
+            tc.cache_variables["FMT_LIB_DIR"] = "lib"
+            if self._has_with_os_api_option:
+                tc.cache_variables["FMT_OS"] = bool(self.options.with_os_api)
```

Comment:
no need for cache_variables, variables is fine, except maybe for `FMT_LIB_DIR`.

Reply:
What about the previous comment that was in the recipe?

```
# FIXME : https://github.com/conan-io/conan/issues/11476
# can be replaced by https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#cache-variables in 1.50
```

Reply:
variables is fine, you are passing defines to build fmt only. Replace cache_variable, so we can merge it soon.

Reply:
I think fmt is already both v1 and v2 compatible, at least we were able to build it in both environments sucessfully

Reply:
partially compatible, yes. This PR improves V2 compatibility. `self.info.header_only()` is now `self.info.clear()`. And, builddirs should be empty.

Reply:
@jwillikers don't forget to replace `cache_variables` by `variables` please

Reply:
@uilianries Thanks for the mention, I missed the responses to this thread. I've fixed that now.